internal/logging/context.go

```diff
@@ -8,27 +8,29 @@ import (
 	testing "github.com/mitchellh/go-testing-interface"
 )
 
-// InitContext creates SDK logger contexts.
+// InitContext creates SDK logger contexts when the provider is running in
+// "production" (not under acceptance testing). The incoming context will
+// already have the root SDK logger and root provider logger setup from
+// terraform-plugin-go tf5server RPC handlers.
 func InitContext(ctx context.Context) context.Context {
-	ctx = tfsdklog.NewRootSDKLogger(ctx)
-	ctx = tfsdklog.NewSubsystem(ctx, SubsystemHelperResource, tfsdklog.WithLevelFromEnv(EnvTfLogSdkHelperResource))
 	ctx = tfsdklog.NewSubsystem(ctx, SubsystemHelperSchema, tfsdklog.WithLevelFromEnv(EnvTfLogSdkHelperSchema))
 
 	return ctx
 }
```

Comment:
Q: Is there a way to verify the pre-condition described in the comment, before letting `InitContext` return? Does `tfsdklog` offer a method to check if the SDK root logger is indeed set?

Reply:
Great question! `tfsdklog` doesn't currently offer a "GetRootSDKLogger" or "HasRootSDKLogger" function. All provider SDK code now is fronted by terraform-plugin-go, which is responsible for creating the root loggers, no matter whether terraform-plugin-framework, terraform-plugin-sdk/v2, and/or terraform-plugin-mux is being used. If the root loggers are not in place, any subsystem logs will be flooded with warning log entries about there not being a root loggers.

terraform-plugin-log code for reference: https://github.com/hashicorp/terraform-plugin-log/blob/94541a862d7fb4b6920239ef9206072c650971fa/tfsdklog/subsystem.go#L100